kotest-assertions/kotest-assertions-shared/src/jvmTest/kotlin/io/kotest/assertions/CluesTests.kt

```diff
@@ -0,0 +1,18 @@
+package io.kotest.assertions
+
+import io.kotest.core.spec.style.FunSpec
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+
+class CluesTests : FunSpec({
+   test("withClue should not fail on coroutine thread switch") {
+      withContext(Dispatchers.Unconfined) {
+         withClue("should not fail") {
+            Thread.currentThread().run { println("withClue block begins on $name, id $id") }
```

Comment:
Does the Dispatchers.Unconfined guarantee it'll switch threads after delay? 
I would add the thread name into a set, and after withContext do an assertion that the set has size 2, to ensure it ran on two threads.

Reply:
That's how i understood [the doc](https://kotlinlang.org/docs/coroutine-context-and-dispatchers.html#unconfined-vs-confined-dispatcher) where it is used as an example for the exact situation:
> The Dispatchers.Unconfined coroutine dispatcher starts a coroutine in the caller thread, but only until the first suspension point. After suspension it resumes the coroutine in the thread that is fully determined by the suspending function that was invoked.

Could add that check, though.

Reply:
Done via 7b69d9757a2abffa063b7a5e06e6987f52503da9.